tests/network/probes.go

```diff
@@ -227,6 +228,28 @@ var _ = SIGDescribe("[ref_id:1182]Probes", func() {
 			Entry("[test_id:TODO]with working Exec probe", createExecProbe(period, initialSeconds, timeoutSeconds, "uname", "-a"), blankIPFamily),
 		)
 
+		Context("guest agent ping", func() {
+			BeforeEach(func() {
+				vmi = libvmi.NewFedora(withMasqueradeNetworkingAndFurtherUserConfig(
+					withLivelinessProbe(createGuestAgentPingProbe(period, initialSeconds)),
+				)...)
+				vmi = tests.VMILauncherIgnoreWarnings(virtClient)(vmi)
+				By("Waiting for agent to connect")
+				tests.WaitAgentConnected(virtClient, vmi)
+				Expect(console.LoginToFedora(vmi)).To(Succeed())
+			})
+
+			It("[test_id:TODO] VM stops when guest agent is disabled", func() {
+				Expect(stopGuestAgent(vmi)).To(Succeed())
+
+				Eventually(func(g Gomega) {
```

Comment:
why do you need to pass `gomega` as input ? afaiu, that pkg is imported without a qualifier, meaning it will be globally available in the file.

Reply:
This reads weird, especially the part about `...Should(Succeed())` since your function does not return anything.

... also not fond of embedding `Expect`s within an `Eventually` block.

I would re-write that helper func to something like this:
```golang
Eventually(func(g Gomega) (*VirtualMachineInstance, error){
    foundVmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
    if err != nil {
        return nil, err
    }
    if !foundVmi.IsFinal() {
        return nil, fmt.Errorf("VMI %s still running", foundVmi.GetName())
    }
}).Should(BeTrue(), ...)
```

Reply:
It is easier to pass a `Gomega` instance to the function in `Eventually`. Then you can call `g.Expect()` on it, and it will not fail the whole test, only the current eventually loop and then it will try again later. 
Here is the documentation, check the few paragraphs at the end.
https://pkg.go.dev/github.com/onsi/gomega#Eventually

Basically this is a special syntax, if you want to check multiple thinks inside the loop.

Reply:
That documentation also mentions “complex systems”, which I interpret as complex tests.
This one is not complex IMO.
There are two disadvantages with this approach:
- It is not commonly used in the code base and is somehow the opposite pattern.
   The existing usages use assertions in Eventually to fail earlier and avoid retry.
- The assertions are mixed into two levels, in the body and outside. Instead, the body can just perform the functionality and allow the Eventually matcher to do the assertion.

Reply:
Just to clarify, this is equivalent (and IMO simpler):

```go
Eventually(func() (*v1.VirtualMachineInstance, error) {
    return virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
}, 2*time.Minute, 1*time.Second).Should(Or(BeInPhase(v1.Failed), HaveSucceeded()))
```

This version has the advantage of showing the actually phase and not just a boolean.

Reply:
Ok, I understand.
I didn't know about matcher `BeInPhase()`, thanks. Using it is of course simpler, I will change the code.